packages/Constants.ts

```diff
@@ -90,4 +92,63 @@ For ${Constants.DISPLAY_NAME} support, visit ${Constants.SUPPORT_LINK}
      * @memberof Constants
      */
     public static readonly HOME_DIR = "~/.zowe";
+
+    public static BASE_CONNECTION_OPTION_GROUP = "Base Connection Options";
+
+    /**
+     * Option used in profile creation and commands for hostname
+     */
+    public static BASE_OPTION_HOST: ICommandOptionDefinition = {
+        name: "host",
+        aliases: ["H"],
+        description: "Host name of service on the mainframe.",
+        type: "string",
+        group: Constants.BASE_CONNECTION_OPTION_GROUP
+    };
+
+    /**
+     * Option used in profile creation and commands for port
+     */
+    public static BASE_OPTION_PORT: ICommandOptionDefinition = {
+        name: "port",
+        aliases: ["P"],
+        description: "Port number of service on the mainframe.",
+        type: "number",
+        defaultValue: 443,
+        group: Constants.BASE_CONNECTION_OPTION_GROUP
+    };
+
+    /**
+     * Option used in profile creation and commands for username / ID
+     */
+    public static BASE_OPTION_USER: ICommandOptionDefinition = {
+        name: "user",
+        aliases: ["u"],
+        description: "User name to authenticate to service on the mainframe.",
+        type: "string",
+        group: Constants.BASE_CONNECTION_OPTION_GROUP
+    };
+
+    /**
+     * Option used in profile creation and commands for password/passphrase
+     */
+    public static BASE_OPTION_PASSWORD: ICommandOptionDefinition = {
+        name: "password",
+        aliases: ["pass", "pw"],
+        description: "Password to authenticate to service on the mainframe.",
+        type: "string",
+        group: Constants.BASE_CONNECTION_OPTION_GROUP
+    };
+
+    /**
+     * Option used in profile creation and commands for rejectUnauthorized setting for connecting to z/OSMF
+     */
+    public static BASE_OPTION_REJECT_UNAUTHORIZED: ICommandOptionDefinition = {
+        name: "reject-unauthorized",
+        aliases: ["ru"],
+        description: "Reject self-signed certificates.",
+        type: "boolean",
+        defaultValue: true,
+        group: Constants.BASE_CONNECTION_OPTION_GROUP
+    };
```

Comment:
Disclaimer: Not something to address now:

The non-SSL request of UPS makes me wonder if an option of --protocol [http | https] might be a candidate for base profile. Such an option exists on the JCLCheck plugin. It was quite helpful to be able to quickly switch the CLI when the server was reconfigured.

Reply:
I agree that being able to influence `--protocol` will prove useful for some users!
Let's try to get it included
@tjohnsonBCM 👉 https://github.com/zowe/zowe-cli/issues/723
:yum: :wink: